collector/mdadm_linux.go

```diff
@@ -158,19 +175,27 @@ func parseMdstat(mdStatusFilePath string) ([]mdStatus, error) {
 		}
 		currentMD = mainLine[0]               // The name of the md-device.
 		isActive := (mainLine[2] == "active") // The activity status of the md-device.
-		personality = mainLine[3]             // The personality type of the md-device.
+		personality = ""
+		for _, possiblePersonality := range mainLine {
+			if raidPersonalityRE.MatchString(possiblePersonality) {
+				personality = possiblePersonality
+			}
+		}
 
 		if len(lines) <= i+3 {
 			return mdStates, fmt.Errorf("error parsing mdstat: entry for %s has fewer lines than expected", currentMD)
 		}
 
-		switch personality {
-		case "raid0":
+		switch {
+		case personality == "raid0":
 			active = int64(len(mainLine) - 4)     // Get the number of devices from the main line.
 			total = active                        // Raid0 active and total is always the same if active.
 			size, err = evalRaid0line(lines[i+1]) // Parse statusline, always present.
-		default:
+		case raidPersonalityRE.MatchString(personality):
 			active, total, size, err = evalStatusline(lines[i+1]) // Parse statusline, always present.
+		default:
+			log.Infof("Personality unknown: %s\n", mainLine)
```

Comment:
Not sure the policy on log messages, but this seems like something someone might want to know about given some of the bug reports and the seeming variability of the data in this file.